oauth2-useragent-core/src/main/java/com/microsoft/alm/oauth2/useragent/UserAgentImpl.java

```diff
@@ -84,6 +85,14 @@ AuthorizationResponse encode(final String methodName, final String... parameters
         provider.augmentProcessParameters(command, classPath);
         // TODO: is this the best way to add our JAR?
         classPath.add(System.getProperty("java.class.path"));
+
+        // Attempt to locate our JAR
+        final ArchiveLocator locator = new ArchiveLocator();
+        final String myClassPath = locator.locateArchive(UserAgentImpl.class);
+        if (myClassPath != null) {

```

Comment:
Under what conditions would `locateArchive()` return `null`?  Based on your tests, even providing a class not in an archive returns the path to the folder in which the corresponding `.class` file is found. (which suggests the name `ArchiveLocator` might not be optimal)

Regardless, do we still need to add the value of `java.class.path`?